src/App/Fossa/Analyze.hs

```diff
@@ -81,9 +71,62 @@ data ScanDestination
   = UploadScan URI ApiKey ProjectMetadata -- ^ upload to fossa with provided api key and base url
   | OutputStdout
 
-analyzeMain :: BaseDir -> Severity -> ScanDestination -> OverrideProject -> Bool -> IO ()
-analyzeMain basedir logSeverity destination project unpackArchives = withLogger logSeverity $
-  analyze basedir destination project unpackArchives
+analyzeMain :: BaseDir -> Severity -> ScanDestination -> OverrideProject -> Bool -> [BuildTargetFilter] -> IO ()
+analyzeMain basedir logSeverity destination project unpackArchives filters = withLogger logSeverity $
+  analyze basedir destination project unpackArchives filters
+
+discoverFuncs ::
+  ( Has (Lift IO) sig m,
+    MonadIO m,
+    Has ReadFS sig m,
+    Has Exec sig m,
+    Has Logger sig m,
+    Has Diag.Diagnostics sig m
+  ) =>
+  -- | Discover functions
+  [Path Abs Dir -> m [NewProject]]
+discoverFuncs =
+  [ Bundler.discover',
+    Cargo.discover',
+    Carthage.discover',
+    Cocoapods.discover',
+    Gradle.discover',
+    Rebar3.discover',
+    Gomodules.discover',
+    Godep.discover',
+    Setuptools.discover',
+    Maven.discover',
+    Leiningen.discover',
+    Composer.discover',
+    Cabal.discover',
+    Stack.discover',
+    Yarn.discover',
+    Npm.discover',
+    Scala.discover',
+    RPM.discover',
+    RepoManifest.discover'
```

Comment:
Have we decided on a name?  Are we sticking with `discover'`?

Reply:
Just renamed to `discover`